git-push: add command (#6328)
tldr-pages/tldr: pages/common/git-push.md

> Push commits to a remote repository.
> More information: <https://git-scm.com/docs/git-push>.

- Send local changes in the current branch to its default remote counterpart:

`git push`

- Send changes from a specific local branch to its remote counterpart:

`git push {{remote_name}} {{local_branch}}`

- Send changes from a specific local branch to its remote counterpart, and set the remote one as the default push/pull target of the local one:

> `git push {{remote_name}} -u {{local_branch}}`

Reviewer: Looking at the man page for `git push`, `-u` doesn't even take an argument:
```
git push [--all | --mirror | --tags] [--follow-tags] [--atomic] [-n | --d
ry-run] [--receive-pack=<git-receive-pack>]
                  [--repo=<repository>] [-f | --force] [-d | --delete] [--prune]
 [-v | --verbose]
                  [-u | --set-upstream] [-o <string> | --push-option=<string>]
                  [--[no-]signed|--signed=(true|false|if-asked)]
                  [--force-with-lease[=<refname>[:<expect>]] [--force-if-include
s]]
                  [--no-verify] [<repository> [<refspec>...]]

```

....so the ordering here is misleading. I suggest putting the `-u` either before or after `{{remote_name}}` and `{{local_branch}}`.

Reviewer 2: I prefer before:
```suggestion
`git push --set-upstream {{remote_name}} {{local_branch}}`
```

Author: ```suggestion
`git push -u {{remote_name}} {{local_branch}}`
```

`git push {{remote_name}} -u {{remote_branch}}`

> - Send changes in a given local branch to the named repository's specified remote branch

Author: ```suggestion
- Send changes in a given local branch to the named repository's specified remote branch:
```

> - Send changes in a given local branch to the named repository's specified remote branch:

Reviewer 3: This doesn't make a lot of sense.

Maybe this will read better
```suggestion
- Send changes from a specific local branch to a specified remote branch:
```
Might also want to include that it changes the name of the branch.

Reviewer: That suggestion sounds good @CleanMachine1

Author: ```suggestion
- Send changes from a specific local branch to a specific remote branch:
```

> - Publish a particular local branch to a remote repository, setting the remote branch name:

Reviewer 3: ```suggestion
- Publish a specifiedlocal branch to a remote repository, replacing the remote branch name:
```

Author: ```suggestion
- Send changes in a given local branch to the named repository's specified remote branch
```

> - Publish a particular local branch t oa remote repository, setting the remote branch name:
> 
> `git push {{remote-name}} {{local_branch}}:{{remote_branch}}`
> 

Author: ```suggestion
- Publish a particular local branch to a remote repository, setting the remote branch name:

`git push {{remote_name}} {{local_branch}}:{{remote_branch}}`

```
